src/views/Benchmark/index.jsx

```diff
@@ -8,27 +8,38 @@ class Benchmark extends Component {
   static propTypes = {
     benchmark: PropTypes.string.isRequired,
     platform: PropTypes.string.isRequired,
+    timeRange: PropTypes.number.isRequired,
   };
 
   state = {
     benchmarkData: {},
   };
 
   componentDidMount() {
-    const { platform, benchmark } = this.props;
-    this.fetchData(platform, benchmark);
+    this.mounted = true;
+    const { platform, benchmark, timeRange } = this.props;
+    this.fetchData(platform, benchmark, timeRange);
   }
 
   componentDidUpdate(prevProps) {
-    const { platform, benchmark } = this.props;
-    if (benchmark !== prevProps.benchmark || platform !== prevProps.platform) {
-      this.fetchData(platform, benchmark);
+    const { platform, benchmark, timeRange } = this.props;
+    if (benchmark !== prevProps.benchmark
+      || platform !== prevProps.platform
+      || timeRange !== prevProps.timeRange
+    ) {
+      this.fetchData(platform, benchmark, timeRange);
     }
   }
 
-  async fetchData(platform, benchmark) {
-    this.setState({ benchmarkData: {} });
-    this.setState({ benchmarkData: await fetchData(platform, benchmark) });
+  componentWillUnmount() {
+    this.mounted = false;
```

Comment:
I don't think we need to use `componentWillUnmount`. Correct me if I got it wrong.

https://reactjs.org/docs/react-component.html#componentwillunmount

componentWillUnmount() is invoked immediately before a component is unmounted and destroyed. Perform any necessary cleanup in this method, such as **invalidating timers**, **canceling network requests**, or **cleaning up any subscriptions** that were created in componentDidMount().

Reply:
I added this to avoid receiving this error in the console:
_Can't call setState (or forceUpdate) on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method._

Should I approach this error differently? I'll try to look more into it.

Reply:
Interesting!
Feel free to see how others out there (e.g. stackoverflow) have fixed this.

Happy to take your fix if you're satisfied with it.

Reply:
I've found a few ways to deal with this, but my current solution seems to carry the lowest cost. Since you can't [really](https://github.com/facebook/react/issues/5465#issuecomment-157888325) cancel es6 Promises, switching to observables or to a state management lib for this reason would be silly, and [isMounted()](https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html) is an antipattern, I think I'll stick to my first solution if it works.

> An easy migration strategy for anyone upgrading their code to avoid isMounted() is to track the mounted status yourself. Just set a _isMounted property to true in componentDidMount and set it to false in componentWillUnmount, and use this variable to check your component’s status.
-- from the same article that's discussing isMounted() as an antipattern

Reply:
👍